ludwig/features/bag_feature.py

```diff
@@ -94,16 +94,15 @@ def add_feature_data(
 
 @register_input_feature(BAG)
 class BagInputFeature(BagFeatureMixin, InputFeature):
-    encoder = "embed"
-    vocab = []
+    encoder = {TYPE: "embed", "vocab": []}
 
     def __init__(self, feature, encoder_obj=None):
         super().__init__(feature)
         self.overwrite_defaults(feature)
         if encoder_obj:
             self.encoder_obj = encoder_obj
         else:
-            self.encoder_obj = self.initialize_encoder(feature)
+            self.encoder_obj = self.initialize_encoder()
```

Comment:
how are parameters passed to the necoder initilization if not through feature?

Reply:
So they are all saved on the object now which is fully populated with the user specified parameters via line 101 in `self.overwrite_defaults(feature)`

Reply:
I believe they are accessed by self.encoder, which is not actually an encoder but an encoder config. In general not a huge fan of functions that refer to internal class state instead of passing the parameters explicitly, as those functions may fail then that internal state is not set

Reply:
I'm seeing the same pattern is true for decoder, so we would want to consider both if we want to make changes to this

Reply:
also do we actually need to keep araound the encoder end decoder configs in self after the encoder and decoder objects are initialized?

Reply:
I agree, and would rather us init the encoder / decoder from the config before we even get to the input / output features. This is one of the major reasons for the move to schema objects.

Reply:
We don't need to retain the encoder configs in the feature after instantiating the encoder, but there are enough places in the code which reference them I'd recommend removing them in a follow-up.